plugins/gtav/gtav.cpp

```diff
@@ -3,118 +3,213 @@
-		 peekProc(avatar_pos_address, avatar_pos_corrector, 12) &&     // Avatar Position values (X, Z and Y).
-		 peekProc(camera_pos_address, camera_pos_corrector, 12) &&     // Camera Position values (X, Z and Y).
-		 peekProc(avatar_front_address, avatar_front_corrector, 12) && // Avatar Front Vector values (X, Z and Y).
-		 peekProc(avatar_top_address, avatar_top_corrector, 12) &&     // Avatar Top Vector values (X, Z and Y).
-		 peekProc(camera_front_address, camera_front_corrector, 12) && // Camera Front Vector values (X, Z and Y).
-		 peekProc(camera_top_address, camera_top_corrector, 12) &&     // Camera Top Vector values (X, Z and Y).
-		 peekProc(player_address, player) &&                           // Player nickname.
-		 peekProc(vehicle_address, vehicle) &&                         // Vehicle name if in a vehicle, empty if not.
-		 peekProc(location_address, location) &&                       // Location name.
-		 peekProc(street_address, street);                             // Street name if on a street, empty if not.
+		 proc->peek(avatarPos_address, avatarPos_corrector, 12) &&   // Avatar Position values (X, Z and Y).
+		 proc->peek(cameraPos_address, cameraPos_corrector, 12) &&   // Camera Position values (X, Z and Y).
+		 proc->peek(avatarDir_address, avatarDir_corrector, 12) &&   // Avatar Front Vector values (X, Z and Y).
+		 proc->peek(avatarAxis_address, avatarAxis_corrector, 12) && // Avatar Top Vector values (X, Z and Y).
+		 proc->peek(cameraDir_address, cameraDir_corrector, 12) &&   // Camera Front Vector values (X, Z and Y).
+		 proc->peek(cameraAxis_address, cameraAxis_corrector, 12) && // Camera Top Vector values (X, Z and Y).
+		 proc->peek(player_address, player) &&                       // Player nickname.
+		 proc->peek(vehicle_address, vehicle) &&                     // Vehicle name if in a vehicle, empty if not.
+		 proc->peek(location_address, location) &&                   // Location name.
+		 proc->peek(street_address, street);                         // Street name if on a street, empty if not.
 
 	// This prevents the plugin from linking to the game in case something goes wrong during values retrieval from
 	// memory addresses.
 	if (!ok)
 		return false;
 
 	// State
-	if (state != 2) {     // If not in-game
-		context.clear();  // Clear context
-		identity.clear(); // Clear identity
+	if (state != 2) {      // If not in-game
+		*contextPtr  = ""; // Clear contextPtr
+		*identityPtr = ""; // Clear identityPtr
 		// Set vectors values to 0.
-		for (int i = 0; i < 3; i++)
-			avatar_pos[i] = avatar_front[i] = avatar_top[i] = camera_pos[i] = camera_front[i] = camera_top[i] = 0.0f;
-
-		return true; // This tells Mumble to ignore all vectors.
-	}
-	/*
-	// Begin context
-	std::ostringstream ocontext;
+		for (int i = 0; i < 3; i++) {
+			avatarPos[i] = avatarDir[i] = avatarAxis[i] = cameraPos[i] = cameraDir[i] = cameraAxis[i] = 0.0f;
+		}
```

Comment:
This can be removed, we already initialize the arrays to 0 at the beginning of the function.